### PR TITLE
[PHP] complete support for form style

### PR DESCRIPTION
**Pull request by thomasphansen**
Current code for the PHP generator doesn't properly support the 'form' style. If we look at other implementations (such as the Javascript one), when _explode_ is **true** and _style_ equals to **form**, the parameter should be expanded so that each key inside the parameter (which should then be an array) would result in a new query parameter. There is a very nice example of this in the accepted answer for [this thread at stackoverflow](https://stackoverflow.com/questions/49582559/how-to-document-dynamic-query-parameter-names-in-openapi-swagger).
So, to make this work, in my humble opinion this code might be changed to:
```
        if (${{paramName}} !== null) {
        {{#style}}
            if('form' === '{{style}}' && is_array(${{paramName}})) {
                foreach(${{paramName}} as $key => $value) {
                    $queryParams[$key] = $value;
                }
            }
            else {
                $queryParams['{{baseName}}'] = ${{paramName}};
            }
        {{/style}}
        {{^style}}
            $queryParams['{{baseName}}'] = ${{paramName}};
        {{/style}}
        }
```
This issue was presented as a comment to PR #3984, which has just being merged. This PR implements this change.

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language: @jebentier (2017/07), @dkarlovi (2017/07), @mandrean (2017/08), @jfastnacht (2017/09), @ackintosh (2017/09) heart, @ybelenko (2018/07), @renepardon (2018/12), @wing328

I'm including a yaml file with an excerpt from the Webshipper API definition, which I'm using successfully after adding these changes to my own fork of openapi-generator.

[webshipper.yaml.gz](https://github.com/OpenAPITools/openapi-generator/files/4285969/webshipper.yaml.gz)

### CODE REVIEW

**Comment by thomasphansen** [comment]
@wing328 I got three failed CI test, but they are all unrelated to PHP - is there anything I should do about it?

**Comment by wing328** [comment]
Those CI failures are not related to this PR.

**Comment by wing328** [comment]
LGTM. Thanks for the PR.

File: modules/openapi-generator/src/main/resources/php/api.mustache [hunk 1]
```diff
@@ -472,11 +472,22 @@ use {{invokerPackage}}\ObjectSerializer;
         $multipart = false;
 
         {{#queryParams}}
-
         // query params
         {{#isExplode}}
         if (${{paramName}} !== null) {
+        {{#style}}
+            if('form' === '{{style}}' && is_array(${{paramName}})) {
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/php/api.mustache]: I may submit a PR later to add a space after `if` and put line 484 in the same line as line 483, i.e. `} else {`

**Comment by wing328** [comment]
@thomasphansen thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952

**Pull request merged**